README.md

```diff
@@ -25,39 +27,52 @@ or
 
-We initially provide a re-implementation of the LLaMA2 architecture. To enable the model to compile, we reimplement `RoPE` encodings without complex numbers. We have verified that the `forward` pass compiles (there is work that needs to be done for `backward` to work with FSDP).
+We provide a re-implementation of the Llama architecture. To enable the model to compile, we reimplement `RoPE` encodings without complex numbers. We have verified that the `forward` pass compiles (there is work that needs to be done for `backward` to work with FSDP).
 
-The figure below shows the latency improvements as we move from eager mode execution to adding SDPA, compile, and SDPA+Compile. The measurements are for 7 and 13B models.
+#### Inference latency
+We measured inference latencies with 1024 token prompt and generation of 256 tokens on AWS P4DN instance nodes with 8 80G A100 GPUs and report the median latency in the below table.
+| Model | # GPUs | Median latency (ms) |
+| ----- | ----------- | ----- |
+| 7B | 1 | 14ms |
+| 13B | 1 | 22ms |
+| 70B | 8 | 30ms |
 
-![image (21)](static/optimizations.png)
+If you would like to reproduce the latencies, you can run the `scripts/benchmark_inference.py` and the details are described in [inference](./scripts).
 
 ## HF Model Support
 
+You can experience the models from Hugging Face APIs using our HF model adapter. The speedups obtained were measured using the Meta checkpoints for the models. One can also obtain similar speedups using the HF adapter with our implementation of the Llama architecture as follows:
+
 ```python
 # fms model
 llama: LLaMA = LLaMA(config)
 
 # huggingface model backed by fms internals
 llama_hf = LLaMAHFForCausalLM.from_fms_model(llama)
 
-# generate some text
+# compile the model -- in HF, the decoder only
+model.decoder = torch.compile(model.decoder)
+
+# generate some text -- the first time will be slow since the model needs to be compiled, but subsequent generations should be faster.
 llama_generator = pipeline(task="text-generation", model=llama_hf, tokenizer=tokenizer)
 llama_generator("""q: how are you? a: I am good. How about you? q: What is the weather like today? a:""")
 ```
 
+A detailed example is provided [here](./notebooks/hf_llama_generation_example.ipynb).
 
-Tensor parallel inference numbers for 13B and 70B models are **coming soon**!
+## Tuning (Coming Soon!!)
 
 ## Training (Coming Soon!!)
 
 ## Open Issues
+There are open issues that we are tracking to improve the stability and memory footprint of the inference latencies.
+
 * https://github.com/pytorch/pytorch/issues/108780 requires adding graph breaks to preserve accuracy.
 * https://github.com/pytorch/pytorch/issues/107824 prevents training/finetuning from working
 
```

Comment:
we need to file the cuda graph issues, but that also requires better repros than what we currently have

Reply:
what can we add here @ani300 ? Any issue pointers?